src/Nethermind/Nethermind.Network.Discovery/DiscoveryApp.cs

```diff
@@ -240,6 +240,12 @@ private void AddPersistedNodes(CancellationToken cancellationToken)
                 break;
             }
 
+            if (!_discoveryManager.NodesFilter.Set(networkNode.HostIp))
```

Comment:
Why not use `Enode` instead of `Ip`?

Reply:
Enode is unique; don't want to repeatedly connect to the same ip/node just on different ports; rather than trying a different node entirely?

Reply:
But those would be different nodes, right? Also can there be multiple nodes on multiple machines under same public IP?

Maybe argument about geographical diversity of connected nodes would be more compelling, but not sure.

Reply:
Potentially same node, different public key or port, but same bandwidth. Better to connect to a different node with different bandwidth when syncing rather than all going to the same ip?